src/other/precompile.jl

```diff
@@ -7,3 +7,10 @@ function _precompile_()
     Base.precompile(Tuple{typeof(pretty),BlueStyle,CSTParser.EXPR,State})
     Base.precompile(Tuple{typeof(pretty),MinimalStyle,CSTParser.EXPR,State})
 end
+using SnoopPrecompile
+@precompile_setup begin
+    dir = joinpath(@__DIR__,"..", "..")
+    @precompile_all_calls begin
+        format(dir)
```

Comment:
would it be more beneficial to run the entire test suite in here? since `format(dir)` would be a subset of the calls in the test suite

Reply:
I added a `format_text` with a few different styles.

I haven't looked through the code extensively, but I wonder if it'd be better to handle styles via [Unityper](https://github.com/YingboMa/Unityper.jl), [Expronicon](https://discourse.julialang.org/t/ann-expronicon-v0-8-type-stable-rust-like-algebra-data-type-pretty-printing-julia-expr-more/92647), or manual `@enum`s?
Manual `@enum`s might be the easiest if you aren't doing many branches or holding information in the structs.

Reply:
information is held in the structs so I don't enum is viable but perhaps one of the of the other options